src/connectors/wire/WireServer.cpp

```diff
@@ -25,15 +28,35 @@ void SocketServer::doListen(basic_socket_acceptor<Protocol, Service>& acceptor,
 
 template <typename Protocol, typename Service>
 void SocketServer::doAcceptOnce(basic_socket_acceptor<Protocol, Service>& acceptor) {
-    typename Protocol::iostream stream;
-    acceptor.accept(*stream.rdbuf());
-    processStream(stream);
+    {
+        boost::shared_ptr<typename Protocol::iostream> stream(
+                boost::make_shared<typename Protocol::iostream>());
+        acceptor.async_accept(*stream->rdbuf(),
+                boost::bind(&SocketServer::processStream,
+                    this, boost::asio::placeholders::error, stream));
+    }
```

Comment:
I missed that you changed this. Why was it needed? Doesn't it make code more complicated?